pkg/segment/reader/segread/multicolreader.go

```diff
@@ -160,6 +161,7 @@ func InitSharedMultiColumnReaders(segKey string, colNames map[string]bool, block
 		numReaders:      numReaders,
 		numOpenFDs:      maxOpenFds,
 		allFDs:          allFDs,
+		ErrorColMap:     make(map[string]error),
```

Comment:
nit: add comment about what is the key for this map : `map[blah] ==> error`

Reply:
this is an unbounded map, not sure what the key is here, but if it is a high cardinality value, then this map could grow high. Suggest adding a limiting whereever this var is used. 
Rather prefer to keep this var unexported and add a method to log the error. That way you can limit how many things can go into this map in one place